Fixes #4774  Leaderboard: Taping on avatar should open tip popup

ratneshnavlakhe opened the pull request:
Fixes #4774 
Setup an on-click listener on avatar image and show alert on click of an avatar with message on how to setup avatar.

ratneshnavlakhe commented:
@misaochan Can you please review this PR.

nicolas-raoul commented on app/src/main/java/fr/free/nrw/commons/profile/leaderboard/UserDetailAdapter.java:
```diff
@@ -84,6 +86,17 @@ public void onBindViewHolder(@NonNull UserDetailAdapter.DataViewHolder holder, i
+        avatar.setOnClickListener(new DebouncingOnClickListener() {
+            @Override
+            public void doClick(final View v) {
+                new AlertDialog.Builder(v.getContext())
```
Could you please add a unit test to cover this code? Thanks!

4D17Y4 replied:
Can you check your dependencies ? I just ran tests and it runs fine. 
@nicolas-raoul can we approve this PR and add tests in another PR ?

nicolas-raoul replied:
Our policy for new pull requests is to require unit tests in the same pull request.

4D17Y4 replied:
Ok, thanks.
@ratneshnavlakhe let us know in case u face other errors. Also if you wish you can commit your draft test and we would be happy to take a look at it.

nicolas-raoul commented on app/src/main/java/fr/free/nrw/commons/profile/leaderboard/UserDetailAdapter.java:
```diff
@@ -84,6 +86,17 @@ public void onBindViewHolder(@NonNull UserDetailAdapter.DataViewHolder holder, i
             holder.getContext().getResources().getString(R.string.count_prefix),
             leaderboardResponse.getCategoryCount()));
 
+        avatar.setOnClickListener(new DebouncingOnClickListener() {
```
Please add a comment explaining what this block of code does. Thanks!

4D17Y4 replied:
Its generally a good idea to have well documented readable code. 
How about adding "Show instructions to set up avatar" as an additional comment ?

nicolas-raoul reviewed (commented):
Otherwise, it is working fine.
The tip shows when tapping on the avatar of someone else (when loading at the leaderboard after tapping someone else) but I don't really consider that as a problem.

misaochan reviewed (approved):
LGTM, happy to merge pending requested changes.

ratneshnavlakhe commented:
Hi, @nicolas-raoul can you please review the comments :)

nicolas-raoul reviewed (changes requested):
Did you address the comments?
Don't forget to commit and push to the same remote branch 🙂

ratneshnavlakhe commented:
Hi @nicolas-raoul Yes I have addressed your comments with a reply, there are some issues that I am facing while running the test case for the Adapter, can you please the stack trace in the above comments.

nicolas-raoul commented:
Unfortunately it seems that your replies did not reach GitHub, as you can see at https://github.com/commons-app/apps-android-commons/pull/4792/files, example:
<img width="817" alt="Screen Shot 2022-02-15 at 18 13 35" src="https://user-images.githubusercontent.com/99590/154030310-166ef118-e037-4090-aa02-b2d9e85a09f7.png">
Could you please try again?

ratneshnavlakhe commented:
It shows pending not sure what I need to do here.

<img width="803" alt="Screenshot 2022-02-15 at 4 19 10 PM" src="https://user-images.githubusercontent.com/4018792/154046983-d01151f1-12f6-4ac5-910b-1f7b49d31f28.png">

nicolas-raoul commented:
It looks like you are trying to review your own pull request.

When you review a pull request, you can send your comments by clicking `Submit review` at the top-right of the page:
<img width="655" alt="Screen Shot 2022-02-15 at 19 52 23" src="https://user-images.githubusercontent.com/99590/154047571-e1f639bd-c468-4d52-b355-7395f14a9da4.png">

neslihanturan commented:
Hi, is this PR still alive?:)

ratneshnavlakhe commented:
Hi @neslihanturan yes, sorry I was not able to contribute to this PR if you want you can pick this up cc @nicolas-raoul

neslihanturan commented:
Thanks @ratneshnavlakhe :)
